### PR TITLE
New features

### PR DESCRIPTION
**Pull request by kiennt**
In this pull request I added:
- Lookup request to get a entity from a key
- KeyUtils module to encode/decode key to/from an urlsafe
  The encoding/decoding algorithm is get from Python ndb library  
  (https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/master/ndb/key.py)

### CODE REVIEW

**Comment by peburrows** [comment]
Thanks for this! Looks like the Travis build failed for some reason, so I'll need to dig in a bit before I merge, but these look like helpful changes.

**Comment by kiennt** [comment]
@peburrows 

I think it is because I use elixir 1.2.5 which has `Base.url_encode64/1` and `Base.url_decode64/1`. 
These functions are not available in elixir 1.2.0.

File: lib/datastore_v1.proto [hunk 1]
```diff
@@ -180,6 +180,14 @@ message Entity {
...
   // Each property's name must be unique for its entity.
   repeated Property property = 2;
+
+  // The entity's properties.
+  // The map's keys are property names.
+  // A property name matching regex `__.*__` is reserved.
+  // A reserved property name is forbidden in certain documented contexts.
+  // The name must not contain more than 500 characters.
+  // The name cannot be `""`.
+  map<string, Value> properties = 3;
```
Comment (peburrows) [inline lib/datastore_v1.proto]: I don't think we want to change the protocol buffer file for the v1beta2 implementation; I pulled ours straight from [the official one](https://github.com/GoogleCloudPlatform/google-cloud-datastore/blob/v1beta2-rev1-3.0.2/proto/datastore_v1.proto), and I think we should stick with it.

When we're ready to upgrade to v1beta3, we have the protobuf file for that in a separate branch, here: https://github.com/peburrows/diplomat/blob/v1beta3/lib/datastore_v1beta3.proto

File: lib/diplomat/key_utils.ex [hunk 2]
```diff
@@ -0,0 +1,200 @@
+defmodule KeyUtils do
```
Comment (peburrows) [inline lib/diplomat/key_utils.ex:1]: Can you help me understand what this is for? It doesn't seem to be used in any of the code (unless I'm missing it).

File: lib/datastore_v1.proto (hunk 1 above)
Comment (kiennt) [inline lib/datastore_v1.proto]: I agreed. This is my bad.

I will revert this one.

File: lib/diplomat/key_utils.ex (hunk 2 above)
Comment (kiennt) [inline lib/diplomat/key_utils.ex:1]: KeyUtils module is for convert a struct Key to urlsafe

I am using Datastore with a library call NDB (this is official library of Google for Python to work on Google AppEngine), and NDB implement a method call `urlsafe` to convert a key to an unique string.

This module is port of `urlsafe` from python to Elixir

Comment (kiennt) [inline lib/diplomat/key_utils.ex:1]: Here is how `urlsafe` is implemented in Python NDB

https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/master/ndb/key.py#L553

Comment (peburrows) [inline lib/diplomat/key_utils.ex:1]: If I understand the python library properly, it seems the urlsafe stuff is just a convenience for initializing a Key from a base64-encoded reference. Is that something that we actually need in an Elixir implementation?

Can you explain the use case for this a bit more? When would you actually need to use this?

Comment (kiennt) [inline lib/diplomat/key_utils.ex:1]: Urlsafe stuff could be use as an unique id for an entity. 
If an entity have not parent, we could use Key.id as unique id, but when entity has parent, 2 entities of same kind could have same id. In that case, I must use `urlsafe` as unique id

I am migrating our application from Python to Elixir, and in Python app, I already use `urlsafe` stuff, so I think it is helpful if the Elixir library also support implement this function.

Comment (peburrows) [inline lib/diplomat/key_utils.ex:1]: Okay, thanks for the info!

**Comment by peburrows** [comment]
@kiennt would you mind changing the Elixir version requirement in the `mix.exs` file to something similar to the following to see if we can get the build to pass (since, as you mentioned, the url encode/decode versions aren't available in `1.2.0`):

``` elixir
elixir: "~> 1.2.5 or ~> 1.3",
```

Once we can get the build to pass, I'll be happy to merge this into master. Thanks again for the contribution!

**Comment by kiennt** [comment]
@peburrows I changed mix file and .travis.yml too to test on elixir 1.2.5 and 1.3

would you mind to take a look again?

**Comment by kiennt** [comment]
@peburrows what do you think about this PR?

I also have another PR to upgrade API to v1beta3, but it is depends on this PR.

**Comment by peburrows** [comment]
Sorry, @kiennt, I've been traveling and am just now seeing your updates. I'll get this merged and push a new release today.

Thanks again!

**Pull request merged**